### PR TITLE
fix: Purge orphaned paths from dynamic path lists in widgets

### LINKED ISSUE #10768: [Bug]:  Menu group button widget entered a bad state
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The app which has Menu group button widget entered a bad state. It created an invisible button reference which caused it to go into cyclic dependency.

Observation: 
1. even if button is removed, it’s path is not removed from triggerPath list from DataTree
2. somehow delete button got added to groupButtons to DataTree after 2 nd button addition

### Steps To Reproduce

1. create group button
2. add on click on button1, button2 and more’s menu item one
3. delete more item(button3)
4. add new button
5. add another button

### Environment

Production

### Version

Cloud

### LINKED ISSUE #10791: Error: Cyclic dependency, node was:"MenuButton.menuItems"
Sentry Issue: [APPSMITH-1Q1](https://sentry.io/organizations/appsmith/issues/2978042617/?referrer=github_integration)

```
Error: Cyclic dependency, node was:"MenuButton.menuItems"
  at None (workers/evaluationUtils.ts:51:8)
```

### PR DESCRIPTION
**Pull request by riodeuno**
## Description

On certain occasions, we find that paths which are defined in a widget's `dynamicTriggerPathList` and `dynamicBindingPathList` don't really exist in the widget. This has been observed in the [MenuButton widget](https://github.com/appsmithorg/appsmith/issues/10768)

This results in issues in subsequent evaluations. The details with respect to the exact issue in the evaluations are unclear -- nevertheless, having orphaned paths is akin to having a bad DSL. This fix purges then during property update.

Fixes #10791
Fixes #10768 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Jest test added for the purging function. 
Tested manually using the steps shown in #10768 

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/purge-orphaned-triggerpaths 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.05 **(0)** | 36.62 **(0.01)** | 34.79 **(0.02)** | 55.45 **(0)**
 :red_circle: | app/client/src/sagas/WidgetOperationSagas.tsx | 51.54 **(0)** | 38.73 **(-0.29)** | 52.08 **(0)** | 52.79 **(0)**
 :green_circle: | app/client/src/sagas/WidgetOperationUtils.ts | 61.86 **(1.08)** | 44 **(2.06)** | 59.62 **(2.48)** | 61.26 **(1.26)**
 :red_circle: | app/client/src/selectors/commentsSelectors.ts | 83.61 **(-1.64)** | 61.76 **(-2.95)** | 73.33 **(0)** | 88.24 **(-2.35)**</details>

### CODE REVIEW

**Review by ApekshaBhosale — APPROVED** [review]

**Review by sbalaji1192 — APPROVED** [review]
LGTM. Let's add some jest cases for purgeOrphanedDynamicTriggerPaths

**Review by ApekshaBhosale — APPROVED** [review]

File: app/client/src/sagas/WidgetOperationUtils.test.ts [hunk 1]
```diff
@@ -581,4 +583,43 @@ describe("WidgetOperationSaga", () => {
...
     expect(result).toStrictEqual(expected);
   });
+
+  it("should purge orphaned dynamicTriggerPaths and dynamicBindingPaths from widget", () => {
+    const input = {
+      dynamicBindingPathList: [
+        { key: "primaryColumns.name.computedValue" },
+        { key: "primaryColumns.name.fontStyle" },
+        { key: "primaryColumns.name.nonExistentPath" },
+        { key: "nonExistentKey}" },
```
Comment (sbalaji1192) [inline app/client/src/sagas/WidgetOperationUtils.test.ts:593]: let's remove `}` at the end. It might be confusing later.

**Review by sbalaji1192 — APPROVED** [review]
LGTM

File: app/client/src/sagas/WidgetOperationUtils.test.ts (hunk 1 above)
Comment (riodeuno) [reply app/client/src/sagas/WidgetOperationUtils.test.ts]: Done

**Pull request merged**